### PR TITLE
style: enable new cops

### PR DESCRIPTION
**Pull request by DirtyF**
This is a 👨🏻‍🎤  style enhancement.

## Summary

- Bump to [Rubocop 1.8.0](https://github.com/rubocop-hq/rubocop/releases/tag/v1.8.0) 🆙 
- Enable new cops 👮🏻 
- Add extensions for minitest, rake and rspec 💅🏻 
- Lint Robucop config file ✅ 
- Auto-correct ✨

### CODE REVIEW

File: test/test_filters.rb [hunk 1]
```diff
@@ -940,12 +940,12 @@ def to_liquid
 
         results = @filter.where(hash, "featured", "true")
         assert_equal 2, results.length
-        assert_equal 9.2, results[0]["rating"]
-        assert_equal 4.7, results[1]["rating"]
+        assert_in_delta(9.2, results[0]["rating"])
```
Comment (ashmaroli) [inline test/test_filters.rb:943]: Can we revert these changes?
I feel that `assert_equal` is more intuitive than `assert_in_delta` ...

File: .rubocop.yml [hunk 2]
```diff
@@ -21,6 +24,7 @@ AllCops:
     - script/**/*
     - vendor/**/*
     - tmp/**/*
+  NewCops: enable
```
Comment (ashmaroli) [inline .rubocop.yml:27]: I'm not so sure we should enable new cops without reviewing their effects first..
CI could fail with an offense unrelated to a commit or PR.

File: .rubocop.yml [hunk 3]
```diff
@@ -127,7 +142,6 @@ Lint/UselessTimes:
 Lint/Void:
   Exclude:
     - lib/jekyll/site.rb
-
```
Comment (ashmaroli) [inline .rubocop.yml:130]: Can we have these separators between Cop departments back? Or will they get removed automatically with autocorrect again?

**Review by ashmaroli — COMMENTED** [review]
Just some minor thoughts.. :+1:  otherwise..

File: .rubocop.yml (hunk 3 above)
Comment (DirtyF) [reply .rubocop.yml:130]: YAMLint 🤷🏻

File: .rubocop.yml (hunk 2 above)
Comment (DirtyF) [reply .rubocop.yml:27]: It was kind of the point fix style issues as early as possible, a PR should embrace our styling guide. No strong opinion on this. It was just easier than to add all Minitest default cops in the Rubocop File, I was hoping it would lead to less maintenance work.

Comment (ashmaroli) [reply .rubocop.yml:27]: > a PR should embrace our styling guide.

I agree. But I'm talking about offenses unrelated to changes in the PR..

Say a contributor has RuboCop 1.8 installed locally and has prepared their pull request after ensuring `script/fmt` reports no offense. However, by the time they *actually opens the pull request* on GitHub, RuboCop has released a v1.9.0 with many new cops that get enabled automatically and report offenses on code not related to the pull request but fails the CI build nevertheless.

File: test/test_filters.rb (hunk 1 above)
Comment (DirtyF) [reply test/test_filters.rb:943]: > Use assert_in_delta if comparing floats. Assertion passes if the expected value is within the delta of actual value.

This is intended to compare floats,  it's what we're doing here, something you have to get used to maybe, but once you know it, it makes the test as intuitive as relying on a generic equal.

**Comment by DirtyF** [comment]
@jekyllbot: merge +dev
